subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/hash_checksum_mgmt.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2018-2021 mcumgr authors
+ * Copyright (c) 2022 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <string.h>
+#include "mgmt/mgmt.h"
+#include "fs_mgmt/hash_checksum_mgmt.h"
+
+static struct hash_checksum_mgmt_group *hash_checksum_mgmt_group_list;
+static struct hash_checksum_mgmt_group *hash_checksum_mgmt_group_list_end;
+
+void hash_checksum_mgmt_unregister_group(struct hash_checksum_mgmt_group *group)
+{
+	struct hash_checksum_mgmt_group *curr = hash_checksum_mgmt_group_list, *prev = NULL;
+
+	if (!group) {
+		return;
+	}
+
+	if (curr == group) {
+		hash_checksum_mgmt_group_list = curr->hc_mg_next;
+		return;
+	}
+
+	while (curr && curr != group) {
+		prev = curr;
+		curr = curr->hc_mg_next;
+	}
+
+	if (!prev || !curr) {
+		return;
+	}
+
+	prev->hc_mg_next = curr->hc_mg_next;
+	if (curr->hc_mg_next == NULL) {
+		hash_checksum_mgmt_group_list_end = curr;
+	}
+}
+
+static struct hash_checksum_mgmt_group *hash_checksum_mgmt_find_group(const char *name)
+{
+	struct hash_checksum_mgmt_group *group;
+
+	/* Find the group with the specified group name */
+	for (group = hash_checksum_mgmt_group_list; group != NULL; group = group->hc_mg_next) {
+		if (strcmp(group->group_name, name) == 0) {
+			break;
+		}
+	}
+
+	return group;
+}
```

Comment:
Merge with `hash_checksum_mgmt_find_handler`, it seems that currently `hash_checksum_mgmt_find_handler` is only null check on return value from calling this.

Reply:
Fixed